### ISSUE TITLE
Enable start-stop logging process

### ISSUE THREAD

**Issue by kiwi-oss** [dmfs/tempomate#22]
Currently, one logs time by "booking" fixed time periods in advance. This forces one to follow this frequency which might not be easy for everyone. One can forget at some points or might have left the keyboard for a while. Having to think about this could create mental load that distracts from the tasks.

As an alternative, I suggest the approach by Timewarrior that could also reduce the number of requests to Jira.

In that approach, you only register locally when you start and stop tasks. When a task is started, it will be stopped automatically when starting another one. At the end of the day, you can make corrections, e.g. when you forgot to start or stop at some point.

Finally, you trigger sending the data to Jira. This can also be done automatically at the start of the next day, week etc.

**Comment by dmfs** [comment]
This sounds like multiple changes. For starters, I've created #24 to add a stop option that adjusts the end of the current work-log to "now".
I'm not sure though, if it makes a difference whether the work-logs are created instantly and adjusted afterwards if necessary or if they are only created in one go once you "commit" to tempo.

**Comment by dmfs** [comment]
Oh, also by creating the work-logs instantly, we're probably able to use Jira/Tempo as source of truth when using the extension on multiple computers, see #15 .

**Comment by dmfs** [comment]
#26 should cover the editing part.

**Comment by dmfs** [comment]
FTR, closing the permanent "Working on ..." notification already updates the work-log and sets the end date-time to "now". So you could already set the default duration to 8 hours and stop whenever you take a break.

**Comment by kiwi-oss** [comment]
> This sounds like multiple changes

I meant this mainly as a feature request to allow the alternative "start-stop-edit-commit" approach that people might be using with other tools. But of course this can be built incrementally.

> Jira/Tempo as source of truth when using the extension on multiple computers

Right, if you commit at the end of the day, you can't easily share the state with other devices. It was never a problem for me because I just work and log on one device.

Such points, however, are the reason why I suggested the alternative approach as an addition, not a replacement for the instantly logging approach.

> closing the permanent "Working on ..." notification already updates the work-log

Unless this is mentioned in the notification, this would be unexpected behaviour for me. In my opinion, it'd be better to make it explicit with a "Stop" button on the notification that updates the log and removes the notification

**Comment by dmfs** [comment]
>>    closing the permanent "Working on ..." notification already updates the work-log
>
> Unless this is mentioned in the notification, this would be unexpected behaviour for me. In my opinion, it'd be better to make it explicit with a "Stop" button on the notification that updates the log and removes the notification

True. I was planning to change that as soon as I figured out how to add a separate button to the notification :-)

**Comment by kiwi-oss** [comment]
I have done this in the past, maybe you can [use the same documentation](https://developer.gnome.org/documentation/tutorials/notifications.html)

**Comment by dmfs** [comment]
I've already tried adding actions. Unfortunately, they are only visible when the notification is shown as a banner. I didn't manage to get them to show up in the calendar popup.

I'll try that again. Maybe I missed something the first time.
